server/api/image.py

```diff
@@ -165,6 +201,9 @@ def metadataGenerator():
                enum=['all', 'images', 'metadata'], default='all')
         .param('filter', 'Filter the images by a PegJS-specified grammar.',
                required=False)
+        .param('imageIds',
+               'A JSON array of up to 100 image IDs, which will be exclusively included.',
+               required=False)
```

Comment:
This is a great candidate for [jsonParam](http://girder.readthedocs.io/en/latest/api-docs.html?highlight=jsonParam#girder.api.describe.Description.jsonParam)'s schema validation. Most of the validation in `_parseImageIds()` could be removed, with the example of the ObjectId check.

Here's an example from Covalic:
https://github.com/girder/covalic/blob/1c4805be1f5c72346220bb6d0dc87a9be14e7354/server/rest/submission.py#L359-L387

Reply:
Agreed, but we'd need to use `autoDescribeRoute` in order for it to have any effect. IMO, `autoDescribeRoute` is much too complicated to understand via its documentation alone (and I don't trust the behavior enough to copy-paste from upstream usages), so I'd rather move to that as part of a larger pass.

Reply:
It makes sense to do it as part of a larger pass. I already used it once in this project to take advantage of upstream JSON validation.
https://github.com/ImageMarkup/isic-archive/blob/5dc99980bf2b1be7f2cadb99b3a4a8b93475a3ee/server/api/image.py#L378